index.js

```diff
@@ -22,9 +22,58 @@ const MODES = [
     "Generic"
 ];
 
+// Helper function to generate the HTML string of a table element containing
+// information read from the cacheIR instruction sequence log JSON.
+function generateCacheIRInstructionTable(cacheIRArray) {
+
+  let cacheIRstr = "";
+  cacheIRstr += '<table class="cacheir-inst-seq-table">'
+  cacheIRstr += '<tbody class="cacheir-inst-seq-tbody">'
+
+  // The JSON of the cacheIR instruction sequence logs contains arrays of
+  // objects. Below gets each element of the array, appends the operation value
+  // to the string of HTML.
+  cacheIRArray.forEach((elt) => {
+    cacheIRstr += '<tr class="cacheir-inst-seq-tr">'
+    cacheIRstr += '<td class="op-td">'+elt.op + '</td>';
+
+    // The value of the args property of the cacheIR instruction sequence JSON
+    // is also an array. Here, for each element of the array, we append the
+    // property and value to  the string of HTML that will be injected into the
+    // page.
+    elt.args.forEach((arg) => {
+      if( arg != {} ) {
```

Comment:
I am not sure what this is supposed to check, but this will always be true, because you are comparing a new object to something else.

Reply:
Some of the args properties of the cacheIR objects were empty objects, so I intended to be checking for empty objects here. I realize now that the if-statements directly below this will screen out empty objects, and checking for empty objects here is redundant. I'll remove this, thanks for pointing this out!